src/fromgif.c

```diff
@@ -649,6 +649,7 @@ load_gif(
         frame->frame_no = 0;
 
         s.img_buffer = s.img_buffer_original;
+        // FIXME verify size is sufficient for headers
```

Comment:
Instead of adding FIXME lines for various problems in the gif decoder, I think removing it entirely is the better move, relying on a dedicated C ABI gif decoder…

Reply:
i'm hesitant to do that without some kind of testing corpus. honestly, i'm hesitant even to merge this without such.